### PR TITLE
Adds /VG/'s temperature gun, making it not-shit

### PR DESCRIPTION
**Pull request by Buggy123**
Shamelessly steals /VG/'s temperature gun(which is actually good), see this image for info:

### CODE REVIEW

**Comment by duncathan** [comment]
TEST YOUR PRS. IF YOU DON'T AT LEAST COMPILE IT BEFORE SUBMITTING, I'M GONNA START CLOSING YOUR SHIT. 

Last warning.

**Comment by Buggy123** [comment]
I tested it.

**Comment by MMMiracles** [comment]
Pretty sure you forgot to commit the DMI for the gun itself.

**Comment by Buggy123** [comment]
Shit.

**Comment by Buggy123** [comment]
There, that should have fixed it.

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -43,26 +43,59 @@
...
 	damage = 0
 	damage_type = BURN
 	nodamage = 1
 	flag = "energy"
-	var/temperature = 100
-
+	temperature = null
+
+/obj/item/projectile/temp/New()
+	spawn(1)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/projectile/special.dm]: uhhh?

File: code/modules/research/designs/weapon_designs.dm [hunk 2]
```diff
@@ -101,7 +101,7 @@
 	id = "temp_gun"
 	req_tech = list("combat" = 3, "materials" = 4, "powerstorage" = 3, "magnets" = 2)
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 5000, MAT_GLASS = 500, MAT_SILVER = 3000)
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 500)
```
Comment (Fox-McCloud) [inline code/modules/research/designs/weapon_designs.dm]: Giving the temp gun this massive of a QoL *and* dropping the silver cost is asking for trouble, IMO.

This means you can mass produce them in autolathes, too.

File: code/modules/projectiles/projectile/special.dm (hunk 1 above)
Comment (Buggy123) [inline code/modules/projectiles/projectile/special.dm]: I don't know, but without that spawn it won't use the correct icon when it fires. Guncode is a confusing mess IMO.

File: code/modules/research/designs/weapon_designs.dm (hunk 2 above)
Comment (Buggy123) [inline code/modules/research/designs/weapon_designs.dm]: That was unintentional, i'll fix it.

**Comment by Buggy123** [comment]
There. The new icon files was considered a untracked file by git, so it wasn't added to the commit. It's fixed now.

**Comment by Buggy123** [comment]
Image of it working:

space station 13 2016-03-13 224610

**Comment by Fox-McCloud** [comment]
:+1: to these; they're very useful for Xenobio--we use them---but I definitely think they should keep the silver requirement, otherwise, they'll be too strong.

**Comment by PKPenguin321** [comment]
not like they'll ever be used since they're not the optimal powergaming guns and they're still gated behind firing pins

**Comment by Buggy123** [comment]
The silver requirement being removed was a unintended side-effect of making all the changes on the wrong git branch and only noticing it at the last second.

**Comment by Buggy123** [comment]
Another picture of it tested and working:
space station 13 2016-03-13 225452

**Comment by Buggy123** [comment]
Video of it tested, and working: https://www.youtube.com/watch?v=vP8IMnuJENo

**Comment by Buggy123** [comment]
Also would anyone be opposed to me putting one of these in xenobio? I think it'd spice things up a bit and give a more sciency alternative for killing slimes.

**Comment by PKPenguin321** [comment]
>Also would anyone be opposed to me putting one of these in xenobio? I think it'd spice things up a bit and give a more sciency alternative for killing slimes.

actually not the worst idea. there is a map edit freeze though

**Comment by Buggy123** [comment]
Oh, right, I forgot about that. In the meantime, after testing it against slimes it's not very effective at killing them, would anyone be opposed to me doing what /VG/ did and making it instakill slimes on it's lowest setting?

**Comment by optimumtact** [comment]
@duncathan I'm not sure what you're getting upset about, travis is here to literally catch these simple commit mistakes.

**Comment by Buggy123** [comment]
And I did test it, thoroughly. Anyone in coderbus can attest that I was asking for help for problems that I could've only found by compiling and testing it. The issue arose when pushing it to origin.

Also, small update: It now kills slimes instantly at settings below 50. It also now functions on non-carbon and non-living mobs, not that that is particularly significant, but the idea of setting corpses on fire was kinda funny so I changed it.

**Comment by AdamElTablawy** [comment]
Do the beams pass through windows?

**Comment by Buggy123** [comment]
No, they're a direct subtype of /obj/item/projectile, so they behave more like bullets really. I could make them pass through windows easily, if you think that'd be a good thing to add.

**Comment by clusterfack** [comment]
>/VG/

**Comment by Razharas** [comment]
Make emagged max temp beam more pronounced compared to just max temp beam, they are kinda hard to distinguish

**Comment by AdamElTablawy** [comment]
Yeah, make them pass through windows so that you can shoot slimes.

**Comment by ToGWtF** [comment]
Yeah they are under special projectiles but should be under energy. It literally is a "Freeze beam" afterall, and that would give it passflags for windows

**Comment by ToGWtF** [comment]
They are literally "Freeze beams". I considered changing them to energy rather than special

**Comment by Shadowlight213** [comment]
Still doubt they'll be used often as we have the xenobio console now. They would be useful against escaped slimes though.

**Comment by Buggy123** [comment]
It's fairly easy to distinguish between emagged and normal temp beams. I mean, we do have practice laser guns which are indistinguishable unless you look at chat, so by comparison a a beam that does nothing but heat you a little VS a bigger orange beam that IGNITES you is pretty easy to distinguish.

Also, they can now pass through things like lasers.

File: code/modules/projectiles/guns/energy/special.dm [hunk 3]
```diff
@@ -270,12 +270,175 @@
...
+	dat += "<A href='?src=\ref[src];temp=-100'>-</A> "
+	dat += "<A href='?src=\ref[src];temp=-10'>-</A> "
+	dat += "<A href='?src=\ref[src];temp=-1'>-</A> "
+	dat += "[target_temperature] "
+	dat += "<A href='?src=\ref[src];temp=1'>+</A> "
+	dat += "<A href='?src=\ref[src];temp=10'>+</A> "
+	dat += "<A href='?src=\ref[src];temp=100'>+</A>"
+	dat += "<BR>"
+	dat += "Power cost: "
+	dat += "<FONT color=[powercostcolor]><B>[powercost]</B></FONT>"
```
Comment (KorPhaeron) [inline code/modules/projectiles/guns/energy/special.dm]: @neersighted send help

**Comment by KorPhaeron** [comment]
Last time we had pinless guns science spammed them and ran around shooting everyone every round, but I guess we will see.

Don't really like adding more emag functions either, but whatever. Temp guns are trash, may as well try this out.

**Comment by Anonmare** [comment]
Traitors can emag test range pins anyway, this is more of a Rev/Cult/Sling/Really desperate antag weapon to hand out as a weak laser gun. It does even less damage than a welder does when not emagged.

**Comment by kyrahabattoir** [comment]
What about the math tho, setting the temperature of an object depends of its mass and what it's made of.

Even if we assume that the "matter" is the same for all living things, the effect should be scaled by mass.

**Comment by Supermichael777** [comment]
but that would be so <i>HARD</i>

**Comment by kyrahabattoir** [comment]
either

target.temperature += beam_energy/target.mass
energy -= beam_energy

or

energy -= temperature*target.mass
target.temperature += temperature

**Comment by Buggy123** [comment]
This also instantly set's the temperature of the ENTIRE mob to exactly the temperature of the beam. Instantly. It doesn't just heat or cool down the part it hits, it heats or cools down the entire mob, completely ignoring any insulating clothing or the low thermal conductivity of flesh.

This isn't the height of accurate science here.

**Comment by duncathan** [comment]
yea and frankly it's dumb

**Comment by optimumtact** [comment]
wow dunc stop rage maintain atmos related stuff again

**Comment by duncathan** [comment]
oranges pls

**Comment by Razharas** [comment]
Proper way would be to make temperature normalization be realistic and slow
Right now humans change temp very fast

**Comment by ToGWtF** [comment]
Bluespace cell AND self-recharging? The max energy usage is only 300 per shot

**Comment by Fox-McCloud** [comment]
Wait..it shouldn't have THAT good of a cell in it.

VG's implementation? You can get off...3? *Maybe* 4 shots with it at max cooling before it's completely depleted.

File: code/modules/projectiles/guns/energy/special.dm [hunk 4]
```diff
@@ -270,12 +270,175 @@
...
-	icon_state = "freezegun"
-	desc = "A gun that changes temperatures."
+	icon = 'icons/obj/gun_temperature.dmi'
+	icon_state = "tempgun_4"
+	item_state = "tempgun_4"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	desc = "A gun that changes the body temperature of its targets."
+	var/temperature = 300
+	var/target_temperature = 300
+	var/e_cost = 100
```
Comment (ToGWtF) [inline code/modules/projectiles/guns/energy/special.dm]: Wouldn't this just cause each shot to cost 100?

Comment (Supermichael777) [inline code/modules/projectiles/guns/energy/special.dm]: this just sets a default so it doesn't runtime if you print one and pull the trigger.

**Comment by Buggy123** [comment]
As requested, I changed it so that it comes with a normal cell instead of a bluespace cell.

**Comment by ToGWtF** [comment]
in-hand sprites?

**Comment by duncathan** [comment]
@Buggy123

**Comment by KorPhaeron** [comment]
@Buggy123 

Let me know if you ever come back/finish this

**Pull request closed without merging**